test/distributed/org/apache/cassandra/distributed/test/guardrails/GuardrailBulkLoadEnabledTest.java

```diff
@@ -0,0 +1,171 @@
+    public void bulkLoaderEnabled() throws Throwable
+    {
+        File sstablesToUpload = prepareSstablesForUpload();
+        // bulk load SSTables work as expected
+        ToolRunner.ToolResult tool = ToolRunner.invokeClass(BulkLoader.class,
+                                                            "--nodes", NODES,
+                                                            "--port", Integer.toString(NATIVE_PORT),
+                                                            "--storage-port", Integer.toString(STORAGE_PORT),
+                                                            sstablesToUpload.absolutePath());
+        tool.assertOnCleanExit();
+        assertTrue(tool.getStdout().contains("Summary statistics"));
+        assertRows(CLUSTER.get(1).executeInternal("SELECT count(*) FROM bulk_load_tables.test"), row(22L));
+
+        // truncate table
+        truncateGeneratedTables();
+        assertRows(CLUSTER.get(1).executeInternal("SELECT * FROM bulk_load_tables.test"), EMPTY_ROWS);
+
+        // Disable bulk load, stream should fail and no data should be loaded
+        CLUSTER.get(1).runOnInstance(() -> Guardrails.instance.setBulkLoadEnabled(false));
+        long mark = CLUSTER.get(1).logs().mark();
+        tool = ToolRunner.invokeClass(BulkLoader.class,
+                                      "--nodes", NODES,
+                                      "--port", Integer.toString(NATIVE_PORT),
+                                      "--storage-port", Integer.toString(STORAGE_PORT),
+                                      sstablesToUpload.absolutePath());
+
+        CLUSTER.get(1).logs().watchFor(mark, "Bulk load is disabled");
+        tool.asserts().failure().errorContains("Stream failed");
+        assertRows(CLUSTER.get(1).executeInternal("SELECT * FROM bulk_load_tables.test"), EMPTY_ROWS);
+
+        // Enable bulk load again, data should be loaded successfully
+        CLUSTER.get(1).runOnInstance(() -> Guardrails.instance.setBulkLoadEnabled(true));
+        tool = ToolRunner.invokeClass(BulkLoader.class,
+                                      "--nodes", NODES,
+                                      "--port", Integer.toString(NATIVE_PORT),
+                                      "--storage-port", Integer.toString(STORAGE_PORT),
+                                      sstablesToUpload.absolutePath());
+        tool.assertOnCleanExit();
+        assertTrue(tool.getStdout().contains("Summary statistics"));
+        assertRows(CLUSTER.get(1).executeInternal("SELECT count(*) FROM bulk_load_tables.test"), row(22L));
+    }
+
+    private static File prepareSstablesForUpload() throws IOException
+    {
+        generateSstables();
+        File sstable_dir = copySstablesFromDataDir("test");
+        truncateGeneratedTables();
+        return sstable_dir;
+    }
+
+    private static File copySstablesFromDataDir(String table) throws IOException
+    {
+        File cfDir = new File("build/test/cassandra/bulk_load_tables", table);
```

Comment:
@Runtian please move this to a temporary directory. Also, if we are going to run this test in a multiplexer (e.g. 500 times in a loop to see if a test is stable), you are not deleting this directory after the test. That means that we are polluting the disk space unnecessarily.